core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -101,15 +101,7 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
           result.failure(exception)
         }
         override def onBlockFetchSuccess(blockId: String, data: ManagedBuffer): Unit = {
-          data match {
-            case f: FileSegmentManagedBuffer =>
-              result.success(f)
-            case _ =>
-              val ret = ByteBuffer.allocate(data.size.toInt)
```

Comment:
The copy behavior was introduced by : https://github.com/apache/spark/pull/2330/commits/69f5d0a2434396abbbd98886e047bc08a9e65565. How can you make sure this can be replaced by increasing the reference count?

Reply:
I don't really understand the point of this initial commit tbh, was there ever a rationale for it ? (I can't find any comments).

I made sure it works by testing it on our dataset (it will indeed crash if the ref count is not incremented).

All 69f5d0a does is transforming a ManagedBuffer (abstract) into the concrete sub-type NioManagedBuffer. There is no real reason to copy the data, as long as you keep track of the reference count